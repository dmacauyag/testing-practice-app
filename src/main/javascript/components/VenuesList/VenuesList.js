import { List } from 'material-ui/List'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'
import React from 'react'
import VenueListItem from '../VenueListItem/VenueListItem'

const VenuesList = (props) => {
  const venueItems = props.venues.map((venue, index) => {
    return (
      <VenueListItem
        key={index}
        venue={venue.venue}
        onVenueSelect={props.onVenueSelect}
      />
    )
  })

  return (
    <MuiThemeProvider>
      <div className="venues-list">
        <List className="venues-list__group">
          {venueItems}
        </List>
      </div>
    </MuiThemeProvider>
  )
}

export default VenuesList
