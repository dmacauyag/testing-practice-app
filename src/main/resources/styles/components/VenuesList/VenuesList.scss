.venues-list__group {
  padding: 0;
}
