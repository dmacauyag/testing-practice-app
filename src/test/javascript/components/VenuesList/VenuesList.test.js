import { shallow } from 'enzyme'
import VenuesList from '../../../../main/javascript/components/VenuesList/VenuesList'
import React from 'react'
import renderer from 'react-test-renderer'

describe('the VenuesList component', () => {
  const onVenueSelectMock = jest.fn()
  const setSelectedVenueStateMock = jest.fn()
  const mockVenues = [
    {
      venue: 'test venue',
      id: 12345,
      onVenueSelect: onVenueSelectMock
    }
  ]

  it('should correctly render the VenuesList component', () => {
    const component = renderer.create(
      <VenuesList
        venues={mockVenues}
        onVenueSelect={setSelectedVenueStateMock}
      />
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
