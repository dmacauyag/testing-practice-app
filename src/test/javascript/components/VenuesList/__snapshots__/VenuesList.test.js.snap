// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the VenuesList component should correctly render the VenuesList component 1`] = `
<div
  className="venues-list"
>
  <div
    className="venues-list__group"
    style={
      Object {
        "muiPrepared": true,
        "padding": "8px 0px 8px 0px",
      }
    }
  >
    <div>
      <span
        className="venues-list__item"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
          }
        }
        tabIndex={0}
      >
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
        >
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "position": "relative",
              }
            }
          />
        </div>
      </span>
    </div>
  </div>
</div>
`;
